### DISCUSSION TITLE
Image layers with a content-length of 0 are being uploaded to Swift without any special errors.

### DISCUSSION THREAD

**Discussion by hyeyoung-leee** [goharbor/harbor#20273] [Q&A] [answered]
Hi,

Image layers with a content-length of 0 are being uploaded to Swift(backend storage of harbor) without any special errors in harbor registry.
Is there any way to know the reason?
Or has there been a similar case?

Thanks.

**Comment by MinerYang** [comment, marked as answer]
Hi @hyeyoung-leee 

It means upload blob by chunks according to distribution-spec
https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pushing-a-blob-in-chunks

**Reply by hyeyoung-leee** [reply]
@MinerYang 

Thank you for your response.

If the layer content-length is 0 and there is no data present in Swift after the image upload has been completed, can I assume the patch process was not executed?

1. Obtain a session ID (upload URL) (POST)
2. Upload the chunks (PATCH)
3. Close the session (PUT)

**Reply by MinerYang** [reply]
It needs dump related request and the registry logs to to further analysis. BTW which tools/clients your are using and how you pushing your images, please provide more details.

**Reply by hyeyoung-leee** [reply]
Thanks.
I will check for more detailed information and get back to you with further questions.
